lib/posix/fs.c

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <errno.h>
+#include <posix/pthread.h>
+#include <posix/unistd.h>
+
+union file_desc {
+	struct fs_file_t file;
+	struct fs_dir_t	dir;
+};
+
+struct posix_fs_desc {
+	union file_desc desc;
+	bool is_dir;
+	bool used;
+};
+
+static struct posix_fs_desc desc_array[CONFIG_MAX_OPEN_FILES];
+
+static struct fs_dirent fdirent;
+static struct dirent pdirent;
+
+#ifdef CONFIG_ERRNO
+static inline void set_pthread_errno(int err)
+{
+	unsigned int key = irq_lock();
```

Comment:
Don't see why this locking is needed.  Surely pthread_self() is threadsafe, and errno_var is an atomic type.

Reply:
Just set `errno`.  `errno` is not a variable, but rather a macro that does the right thing.